### PR TITLE
use okhttp3

### PR DESCRIPTION
**Pull request by PatriceJiang**
use `okhttp` to replace `android-async-http` in CCDownloader android implementation. 

base on https://github.com/cocos-creator/cocos2d-x-lite/pull/1731

### CODE REVIEW

File: cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxDownloader.java [hunk 1]
```diff
@@ -273,122 +82,210 @@ public void run() {
...
+                                        tempFile.renameTo(finalFile);
+                                    } while (false);
+
+                                    if (errStr == null) {
+                                        downloader.onFinish(id, ErrorCode.ERROR_NO_ERROR, null, null);
+                                    }
+                                    else
+                                        downloader.onFinish(id, ErrorCode.ERROR_IMPL_INTERNAL, errStr, null);
+                                } else {
+                                    // 非文件
```
Comment (minggo) [inline cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxDownloader.java:253]: comments in chinese

**Comment by halx99** [comment]
why not use curl for all platforms?

**Comment by minggo** [comment]
@halx99 according the issue: https://github.com/cocos2d/cocos2d-x/issues/10382, i think it is about to reduce package size. Use platform network can reduce the package size. But it is pity that it brings JAR lib on Android too, though the package size is smaller than libcurl(curl depends on ssl).

So i think it will be better if not 3rd party library is needed.

**Comment by PatriceJiang** [comment]
re-commit in https://github.com/cocos2d/cocos2d-x/pull/19766

**Pull request closed without merging**